### PR TITLE
fix(card-group): fix offset and optional gradient

### PR DESCRIPTION
**Pull request by ariellalgilmore**
### Related Ticket(s)

#6016 

### Description

added offset option by adding empty to property to card-group-item. CSS custom property for gradient value in outline. 

Talked with Shixie and gradient will not be demo'd in storybook yet. 

### Changelog

**New**

- borderAdjustments, and  _fillLastRowWithEmptyCards 
- added update to call resizeHandler when properties change in gridMode

**Changed**

- resizeHandler now calls _setSameHeight, _borderAdjustments, and  _fillLastRowWithEmptyCards

**Removed**

- {{removed thing}}

### CODE REVIEW

File: packages/styles/scss/components/card-group/_card-group.scss [hunk 1]
```diff
@@ -116,7 +113,21 @@
...
+  //    width: 100%;
+  //    height: 100%;
+  //    position: absolute;
+  //    left: 0;
+  //    top: 0;
+  //    pointer-events: none;
+  //    box-shadow: inset 0 0 0 1px $ui-background;
+  //  }
+  //}
+
```
Comment (kennylam) [inline packages/styles/scss/components/card-group/_card-group.scss:130]: @ariellalgilmore Looks like you forgot to remove these.

**Comment by shixiedesign** [comment]
Looks great! only noticed `same height util` is having a bit of trouble. I was playing with the knobs. Could this be a non-issue if the cards are loaded with preset numbers, then it's okay I think. 
image

**Comment by ariellalgilmore** [comment]
@shixiedesign seems like a storybook thing. When you open up the knob changes in a new tab you can see it being displayed correctly.

**Review by shixiedesign — APPROVED** [review]
Looks great!!! 🔥

**Review by kennylam — APPROVED** [review]
LGTM thanks @ariellalgilmore!

**Pull request merged**